src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -104,28 +107,35 @@ void rotate() throws IOException {
 
     private void finishFile() throws IOException {
         if (isDirty()) {
-            gzipStream.finish();
+            if(compressionType == null){
+                ((GZIPOutputStream) outputStream).finish();
```

Comment:
we can clean this up a bit
1. the compression type is not what matters, but rather the typeof outputstream
2. we should try and maintain the same interface, maybe we can change the gzipoutput stream to flush?

Reply:
flush is not the same as finish . 
although the comment in the next line suggests theres a problem with closing the file it seems to work well